### ISSUE TITLE
Cannot find shared library on cargo test --doc

### ISSUE THREAD

**Issue by termoshtt** [rust-ndarray/ndarray-linalg#229]
https://github.com/rust-ndarray/ndarray-linalg/pull/227/checks?check_run_id=902046181

```
running 7 tests
test src/krylov/mod.rs - krylov::Orthogonalizer (line 51) ... FAILED
test src/cholesky.rs - cholesky (line 12) ... FAILED
test src/solve.rs - solve (line 31) ... FAILED
test src/least_squares.rs - least_squares (line 28) ... FAILED
test src/solve.rs - solve (line 7) ... FAILED
test src/solveh.rs - solveh (line 10) ... FAILED
test src/solveh.rs - solveh (line 35) ... FAILED

failures:

---- src/krylov/mod.rs - krylov::Orthogonalizer (line 51) stdout ----
Test executable failed (exit code 127).

stderr:
/tmp/rustdoctestDSRkP2/rust_out: error while loading shared libraries: libmkl_intel_lp64.so: cannot open shared object file: No such file or directory
```

**Comment by termoshtt** [comment]
This will be fixed in #204 by linking statically

**Comment by termoshtt** [comment]
This has been fixed by #204, which also reverts #231

**Comment by olson-sean-k** [comment]
Thanks for working on this and implementing a fix! I've been seeing these errors in my continuous integration and local doctests since Rust 1.45 entered beta. Do you have a guess for when these changes might be released?
